### PR TITLE
Inform the user why it "Could not remove build area"

### PR DESCRIPTION
**Pull request by jsmeix**
* Type: **Enhancement**

* Impact: **Normal**

* Reference to related issue (URL):
https://github.com/rear/rear/issues/2908

* How was this pull request tested?
Not tested by me.

* Brief description of the changes in this pull request:

In output/ISO/Linux-i386/700_create_efibootimg.sh
when umounting the EFI virtual image fails it is no real error
so only inform the user about the umounting failure
so he can understand why later
cleanup_build_area_and_end_program() shows
```
Could not remove build area
```

### CODE REVIEW

**Comment by jsmeix** [comment]
Primarily for my own information
here some links about 'umount --lazy' and related things:

https://superuser.com/questions/412109/lazy-umount-or-unmounting-a-busy-disk-in-linux

https://stackoverflow.com/questions/7878707/how-to-unmount-a-busy-device

https://serverfault.com/questions/1101277/how-does-a-lazy-unmount-actually-work

https://unix.stackexchange.com/questions/385645/how-do-i-know-when-lazy-umount-l-completes

https://www.unix.com/linux/119194-umount-l-dangerous.html

https://unix.stackexchange.com/questions/111779/how-to-find-out-easily-whether-a-block-device-or-a-part-of-it-is-mounted-someh

**Comment by jsmeix** [comment]
@pcahyna 
in my current change here for
output/ISO/Linux-i386/700_create_efibootimg.sh
I call only plain 'umount --lazy' without additional '--force'
so I cannot use the umount_mountpoint_lazy() function here
because that calls `umount $v -f -l $mountpoint`.

I do not like '--lazy' with additional '--force'
because "man umount" tells:
```
-f, --force
Force an unmount (in case of an unreachable NFS system).
Note that this option does not guarantee that umount command does not hang.
```
As usual the description is rather terse and does not explain things
so one has to imagine what could be meant.

Currently I understand this as if '--force' may hang up
because in general to enforce something one must try hard
to get that something done which means "don't give up"
but "try hard endlessly" so it may never finish.

In contrast '--lazy' has opposite meaning
so 'umount --lazy' basically always returns and
lets things clean up on its own in the background,
e.g. see
https://serverfault.com/questions/1101277/how-does-a-lazy-unmount-actually-work
that reads (excerpt):
```
Lazy umount is a regular umount, which happens in background.
In addition is hides the mount entry, which makes it looks
like umount have worked (or Detach the filesystem from
the file hierarchy, how man page says). In really
the unmount will happen when filesystem is not in use anymore.

Thus, for a real umount one should use --force and wait ...
```

So I think '--lazy' and '--force' contradict each other.

Or what do I misunderstand here?

**Comment by pcahyna** [comment]
@jsmeix I am not sure myself. The impression I get from the manual page is that the main motivation for umount options is the case of NFS mount where the server became unreachable. I am not even sure whether `-f` does anything useful in the case of local filesystems.

**Comment by schlomo** [comment]
We still do `umount ... || LogPrintError ...` so that `rear` will happily continue even in case of an `umount` error.

I'm wondering if we maybe should be more strict and actually `StopIfError` to avoid follow-up problems like reported by @thomas-merz originally? That way he would have noticed the problem via a direct error message from ReaR instead of via some other monitoring noticing lots of new filesystems...

Also, maybe the discussed `sleep 1` would also be a good solution instead of a cascae of `umount` attempts?

I find the `fuser` really useful in case of umount troubles, great idea!

And finally, maybe `umount` should be converted into a ReaR function to always have that error handling attached to it?

**Comment by jsmeix** [comment]
It is my goal behind to get in the end a
sufficiently simple and reliably working
umounting function in ReaR.

But this will take some time in particular
to get sufficient experience how things behave
for real users out there on their systems.

What "just works" for us on our clean test systems
may terribly fail for users on problematic systems.

E.g. see https://github.com/rear/rear/issues/2908
As far as I can remember we never had an issue
like that with plain and simple 'umount' before.

E.g. see my comment in this pull request code
how 'fuser' must be called to avoid possibly
scaring and misleading output in certain cases.

Bottom line:
RFC 1925 item (8) It is more complicated than you think.

That one should even better read:
In practice it is more complicated than one imagines in theory.

File: usr/share/rear/output/ISO/Linux-i386/700_create_efibootimg.sh [hunk 1]
```diff
@@ -44,7 +44,41 @@ mount $v -o loop -t vfat $TMP_DIR/efiboot.img $TMP_DIR/efi_virt || Error "Failed
...
+        # The -M option avoids that fuser may show all processes using the '/' filesystem
+        # ( $TMP_DIR is $BUILD_DIR/tmp which is /var/tmp/rear.XXXXXXXXXXXXXXX/tmp/ )
+        # when $TMP_DIR/efiboot.img got umounted just before fuser starts, see "man fuser":
+        #   The mount -m option will match any file within the same device as the specified file,
+        #   use the -M option as well if you mean to specify only the mount point.
+        # So when $TMP_DIR/efiboot.img is umounted 'fuser -v -M -m $TMP_DIR/efi_virt' only shows
+        #   "Specified filename /var/tmp/rear.XXXXXXXXXXXXXXX/tmp/efi_virt is not a mountpoint"
+        # instead of all processes using '/' (or /var/ or /var/tmp/ if one is a mountpoint)
+        # which would be misleading information that may even look scaring and cause false alarm:
+        fuser -v -M -m $TMP_DIR/efi_virt 1>&2
```
Comment (pcahyna) [inline usr/share/rear/output/ISO/Linux-i386/700_create_efibootimg.sh:69]: See https://github.com/rear/rear/issues/2908#issuecomment-1429605894 - older systems do not support `-M` - but since the code path where it is used is exceptional and the output is used only for information purposes, we don't care that much?

Comment (schlomo) [reply usr/share/rear/output/ISO/Linux-i386/700_create_efibootimg.sh:69]: `fuser` will probably fail if `-M` is not supported, which will hide the actual result that we are interested in.

Maybe `lsof` is more portable?

Comment (jsmeix) [reply usr/share/rear/output/ISO/Linux-i386/700_create_efibootimg.sh:69]: I would like to keep our currently tested 'fuser ... -M ...'
and I agree with the @pcahyna reasoning in his above
https://github.com/rear/rear/pull/2909#discussion_r1105698584

See also
https://github.com/rear/rear/issues/2908#issuecomment-1476124509

**Comment by jsmeix** [comment]
@rear/contributors 
I would like to merge it tomorrow afternoon
unless there are objections.

This is only a first step towards a (hopefully) sufficiently
simple and reliably working umounting function in ReaR, cf.
https://github.com/rear/rear/pull/2909#issuecomment-1429530985

**Comment by pcahyna** [comment]
jsmeix with the addition of `sleep` it helps with #2908, right?

**Comment by jsmeix** [comment]
Yes, the "sleep 1 and retry" helps in particular with
issues like https://github.com/rear/rear/issues/2908
but in general I prefer to try simple things first,
see my new comments via my recent
https://github.com/rear/rear/pull/2909/commits/d6559072f4bc64c5ba18ff2fdd03b410ad340ace

When normal umount works after simple 'sleep 1'
it avoids in particular possible issues with 'fuser -M'
on older Linux distributions that do not support '-M'.

**Comment by jsmeix** [comment]
Via https://github.com/rear/rear/pull/3408
basically a copy of the code in this pull request
is now moved to lib/global-functions.sh as
new umount_mountpoint_retry_lazy function
which is called instead of the code in this pull request
in output/ISO/Linux-i386/700_create_efibootimg.sh
and also in output/USB/Linux-i386/100_create_efiboot.sh
see https://github.com/rear/rear/issues/3397
